src/utility.cpp

```diff
@@ -515,106 +558,128 @@ BOOL PatchImport (HMODULE importmodule, moduleentry_t *patchModule)
 	LPSTR pszBuffer = cwBuffer;
 	DWORD dwMaxChars = _countof(cwBuffer);
 	DWORD dwLength = ::GetModuleFileNameA(importmodule, pszBuffer, dwMaxChars);
-#endif
+#endif
+
+    // have a stack local array of the addresses, don;t want to use malloc for this
+    LPVOID realAddresses[MAX_PATCH_ENTRY_COUNT];
+    patchentry_t* patchEntry = patchModule->patchTable;
+    int i = 0;
+    while (patchEntry->importName)
+    {
+        LPCSTR importname = patchEntry->importName;
+
+        // Get the *real* address of the import. If we find this address in the IAT,
+        // then we've found the entry that needs to be patched.
+        LPVOID import = VisualLeakDetector::_RGetProcAddress(exportmodule, importname);
+        if (!import)
+            import = GetProcAddress(exportmodule, importname);
+        import = FindRealCode(import);
+
+        realAddresses[i] = import;
+
+        patchEntry++;
+        i++;
+
+        if (i >= MAX_PATCH_ENTRY_COUNT)
+        {
+            // we only process MAX_PATCH_ENTRY_COUNT, if we exceed it crash
+            // if this abort is ever hit, it means that MAX_PATCH_ENTRY_COUNT should be bumped up
+            abort();

```

Comment:
>abort(); [](start = 12, length = 8)

Might be nice to log it with `Report()`

Reply:
Added

---
In reply to: [591986587](https://github.com/Azure/vld/pull/8#discussion_r591986587) [](ancestors = 591986587)